src/domain/core/src/messages/core_message_types.rs

```diff
@@ -74,3 +82,11 @@ pub struct DatasetLifecycleMessageDeleted {
 }
 
 ////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub struct DatasetLifecycleMessageRenamed {
+    pub dataset_id: DatasetID,
+    pub new_name: DatasetName,
```

Comment:
Consider including the old name and the owner account ID+name in the message. The place that performs the renaming will likely have all this info on hand, and including it in the message both simplifies diagnostics and is likely to make the life of consumers much easier and avoid additional lookup.

Reply:
Weird, don't know why github was showing me old version of the code for review. This message is gone now, so I will resolve the comment.